Sources/PowerDiagram/PowerDiagram.swift

```diff
@@ -0,0 +1,72 @@
+import class Foundation.FileHandle
+import StringWidth
+import struct Foundation.Selector
+
+public struct PowerDiagram {
+  public struct Label: Comparable {
+    public let value: String
+    public let column: Int
+
+    public init(value: String, column: Int) {
+      self.value = value
+      self.column = column
+    }
+
+    static public func == (lhs: Self, rhs: Self) -> Bool {
+      lhs.column == rhs.column
+    }
+
+    static public func < (lhs: Self, rhs: Self) -> Bool {
+      lhs.column < rhs.column
+    }
+  }
+
+  private let mainLine: String
+  private let labels: [Label]
+
+  public init(mainLine: String, labels: [Label] ) {
+    self.mainLine = mainLine
+    self.labels = labels
+  }
+
+  public func render() -> String {
+    var labels = self.labels.sorted()
+
+    var message = "\(mainLine)\n"
+    renderFirstLine(into: &message)
+
+    while !labels.isEmpty {
+      var current = 0
+      var index = 0
+      while index < labels.count {
+        if index == labels.count - 1
+            || ((labels[index].column + stringWidth(labels[index].value) < labels[index + 1].column)
+                && labels[index].value.unicodeScalars.allSatisfy(\.isASCII))
```

Comment:
`.allSatisfy(\.isASCII)` has the same semantics as `.filter({ !$0.isASCII }).isEmpty`, but it:

1. Spares an Array allocation
2. Skips an Array copy
3. Returns early on the first offending character